src/learn/dotnet-two/index.md

```diff
@@ -0,0 +1,297 @@
+---
+title: Dependency injections and interfaces in Dotnet
+description: Learn how to use dependency injections and interfaces in Dotnet
+tags:
+  - workshop
+keywords:
+  - DOTNET
+  - ASP.NET 
+  - CSharp
+challenge: https://github.com/foundersandcoders/DOTNET-Workshop-Two/blob/main/README.md
+---
+
+
+
+## Setting up .Net
+
+Dot net projects can be made in VSCode just like javascript. In order to get started with this you will need to first install the dot net SDK https://dotnet.microsoft.com/en-us/download and the C# Dev kit https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit
+
+
+
+
+## Why Dependency Injections
+
+It is quite common to have an object that depends on another object to do its function especially in a object focused environment like dotnet. For instance your constructor depends on your data context object. However it can cause problems to always have to build dependencies first and then the objects that depend on them next. Also what if two classes need to access different versions of the dependency for example we might want to use the data context class both in a testing class and in the actual project but when using it in the testing context might want to set it up in such a way that we don't actually effect the database we could do this by adjusting the controller class but it would be easier if we could make that class initially more flexible. 
+
+The solution to these problems is dependency injections where rather than creating a dependency that a class depends on we give the class a interface to tell it what type of object to expect and then create a service that allows us to create dependencies as needed and inject them into the class. 
+
+## Set up
+
+Open the simple hello world project you made last week. Within MyWebApi (or whatevery you named the project) add a models directory and create a Book.cs file within it that looks like this .
+
+```cs
+namespace MyWebApi.Models
+{
+
+    public class Book
+    {
+        public Guid Id {set; get;}
+        public string? Name {set; get;}
+        
+
+    }
+}
+
+```
+We now need to install Microsoft EntityFrameworkCore to help set up a quick in memory database. Like so
+
+
+```dotnet
+
+dotnet add package Microsoft.EntityFrameworkCore
+
+```
+then we make a data folder and inside make a BookContext.cs file that will look like so
```

Comment:
there's a few capitalisation errors and other small editing things. I think the easiest thing is to get ai to correct it. here it is!

https://ivan-takes-notes.notion.site/NET-PR-edits-1b027250634f4b7b8fe2f71e8145b838?pvs=4